pkg/uploadproxy/uploadproxy.go

```diff
@@ -169,9 +172,28 @@ func (app *uploadProxyApp) handleUploadRequest(w http.ResponseWriter, r *http.Re
 
 	klog.V(1).Infof("Received valid token: pvc: %s, namespace: %s", tokenData.PvcName, tokenData.Namespace)
 
+	err = app.uploadPossible(tokenData.PvcName, tokenData.Namespace)
+	if err != nil {
+		klog.Error(err)
+		w.WriteHeader(http.StatusServiceUnavailable)
+		return
+	}
+
 	app.proxyUploadRequest(tokenData.Namespace, tokenData.PvcName, w, r)
 }
 
+func (app *uploadProxyApp) uploadPossible(pvcName, pvcNamespace string) error {
```

Comment:
Should also return error if the pod phase annotation is `Succeded`

Reply:
done